c10/test/util/intrusive_ptr_test.cpp

```diff
@@ -1653,6 +1653,15 @@ TEST(WeakIntrusivePtrTest, givenPtr_whenLocking_thenReturnsCorrectObject) {
   EXPECT_EQ(var.ptr.get(), locked.get());
 }
 
+TEST(WeakIntrusivePtrTest, expiredPtr_whenLocking_thenReturnsNullType) {
```

Comment:
Test looks good but I don't think it has something to do with the bug you fixed. Afaik, expiring the object doesn't set the weak pointer to null but it still points to the expired object. Let's add a test that has a weak nullptr and tries to lock it.

Reply:
I think `intrusive_ptr.reset()` will set `target_ = NullType::singleton();`, and weak_intrusive_ptr is actually sharing the same `TTarget*` with intrusive_ptr as it constructs from it? so weak_intrusive_ptr should be null as well. But either way, I added a explicit weak nullptr test to make sure it works :)